relay-server/src/actors/envelopes.rs

```diff
@@ -1915,8 +1934,9 @@ impl Handler<QueueEnvelope> for EnvelopeManager {
 /// metrics.
 struct HandleEnvelope {
     pub envelope: Envelope,
-    pub project: Addr<Project>,
-    pub sampling_project: Option<Addr<Project>>,
+    pub public_key: ProjectKey,
```

Comment:
What's the reason we call this `public_key` rather than `project_key` everywhere?

Reply:
It was called like that everywhere, I thought about calling it project_key but left it public_key for consistency... we could have probably used `project_public_key`

Reply:
It's even worse, because this inconsistency goes through all of Sentry, too. Originally, there used to be a public and private key for each DSN, but then the public key was referred to as ProjectKey. Today, there is no more secret key, and the public key uniquely identifies each project, acting as a project key. In Relay, wherever we can, we should call it `project_key` to distinguish it from Relay's own public key.

Reply:
ok... I'm happy to refactor, it was a think that slightly bothered me but I originally left it for consistency.
Will refactor to `project_key` wherever is newly added in this PR.